src/ssl/PeekingPeerConnector.cc

```diff
@@ -81,10 +81,10 @@ Ssl::PeekingPeerConnector::checkForPeekAndSplice()
     Security::SessionPointer session(fd_table[serverConn->fd].ssl);
     BIO *b = SSL_get_rbio(session.get());
     Ssl::ServerBio *srvBio = static_cast<Ssl::ServerBio *>(BIO_get_data(b));
-    if (!srvBio->canSplice())
-        acl_checklist->banAction(Acl::Answer(ACCESS_ALLOWED, Ssl::bumpSplice));
-    if (!srvBio->canBump())
+    if (request->clientConnectionManager->serverBump()->act.step2 == Ssl::bumpPeek)
         acl_checklist->banAction(Acl::Answer(ACCESS_ALLOWED, Ssl::bumpBump));
+    else
+        acl_checklist->banAction(Acl::Answer(ACCESS_ALLOWED, Ssl::bumpSplice));
```

Comment:
> I am worried about this change. We are not simply removing disabled functionality here. We are changing how Squid works in certain special cases AFAICT.

I agree here. My goal is to move the decisions about canBump or canSplice from ServerBio to PeekingPeerConnector. After removing the adjustSSL function the ServerBio does not have any other parts required to take a such decision.
But I agree  it is better to not make such changes here.

The related changes removed with commits cbacb45d9 and d9b8bbf1

> P.S. I am also worried that request->clientConnectionManager might become invalid before we reach this context, crashing Squid.

True. To solve this problem I started using a new PeekingPeerConnector::currentBumping mode member and then realized that there is a bigger problem because of using the ConnStateData::sslBumpMode (which also modified by PeekigPeerConnector). But this patch moved to is own branch/PR and this branch now was buggy!